#! /usr/bin/env python

PACKAGE='controllers'

import roslib;
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

fnn_enum = gen.enum([	gen.const("none", 	int_t, 	0, 	"No FNN"),
			gen.const("SMC", 	int_t, 	1, 	"Sliding Mode Control"),
			gen.const("LM", 	int_t, 	2, 	"Levenberg-Marquart")],
			"FNN type")

#       	Name		Type      	Level	Description		Default 	Min	Max 	Values
gen.add( 	"fnn_type", 	int_t, 		0, 	"FNN type", 		1,		0,	2,	edit_method=fnn_enum)
gen.add( 	"alpha",	double_t, 	0, 	"Learning rate", 	0.001, 		0, 	1)
gen.add( 	"gamma",	double_t, 	0, 	"gamma factor", 	1.0, 		0, 	1)
gen.add( 	"nu",		double_t, 	0, 	"nu factor", 		0.8, 		0, 	1)

exit(gen.generate(PACKAGE, "controllers", "setFNN"))
